lib/fog/core/service.rb

```diff
@@ -163,6 +163,8 @@ def coerce_options(options)
           value_string = value.to_s.downcase
           if value.nil?
             options.delete(key)
+         elsif key.end_with? 'name'
+            options[key] = value.to_s
```

Comment:
I'm afraid this behaviour might break other providers that rely on this, no?

Reply:
One provider I use can have a "123" project name currently, and will reject the `123` integer despite filling `"123"` string as option.

I might restrict the check to `elsif value.is_a?(String) && key.end_with? 'name` to be sure, what do you think?

Reply:
Since this is an openstack issue i rather make this change for it only since we have several dozens. WDYT?

Reply:
Yeah, this has the potential for a much broader impact than we really desire. I would suggest we try to make a change in open-stack itself, where we could explicitly check for integer values in project name and raise a specific error. Does that make sense?